drivers/volume/portworx/schedops/k8s-schedops.go

```diff
@@ -179,10 +180,32 @@ func (k *k8sSchedOps) ValidateVolumeSetup(vol *volume.Volume) error {
 		return err
 	}
 
+	nodes := node.GetNodesByName()
+
 	for _, p := range pods {
 		if ready := k8s.Instance().IsPodReady(p); !ready {
 			continue
 		}
+
+		currentNode, nodeExists := nodes[p.Spec.NodeName];
+		if  !nodeExists {
+			return fmt.Errorf("node %s for pod [%s] %s not found", p.Spec.NodeName, p.Namespace, p.Name)
+		}
+
+		// ignore error when a command not exactly fail, like grep when empty return exit 1
+		connOpts := node.ConnectionOpts{
+			TimeBeforeRetry: defaultRetryInterval,
+			Timeout: defaultTimeout,
+			IgnoreError: true,
+		}
+
+		out, _ := d.RunCommand(currentNode,
+			fmt.Sprintf("mount | grep %s", pvName), connOpts)
```

Comment:
Just check the pv name is not sufficient since the path could be mounted on a non-px device and we need to catch that.

So do the same checks (pxd, pxfs etc) that we do for checking the mounts inside the container.